src/dftbp/timedep/timeprop.F90

```diff
@@ -1545,12 +1598,23 @@ subroutine updateH(this, H1, ints, H0, speciesAll, qq, q0, coord, orb, potential
         call adjointLowerTriangle(H1(:,:,iKS))
       end if
     end do
+  #:endif
 
     ! add hybrid xc-functional contribution
     if (this%isHybridXc) then
     #:if WITH_MPI
-      @:RAISE_ERROR(errStatus, -1, "Timeprop Module: MPI-parallelization not implemented for hybrid&
-          & xc-functionals.")
+      deltaRho = rho
+      if (this%nSpin > 2) then
+        @:RAISE_ERROR(errStatus, -1, "HybridXc: Not implemented for non-colinear spin.")
+      end if
+      call denseSubtractDensityOfAtomsCmplxNonperiodicBlacs(env, this%parallelKS, q0, this%denseDesc, deltaRho)
+
+      do iSpin = 1, this%nSpin
+        HSqrCplxCam(:,:) = (0.0_dp, 0.0_dp)
+        call hybridXc%addCamHamiltonianMatrix_cmplx_blacs(env, this%denseDesc, sSqr(:,:,iSpin),&
+            & deltaRho(:,:,iSpin), HSqrCplxCam)
```

Comment:
@vanderhe could you please check if these calls should be fine for the a colinear spin-polarised case? This is the only test that is failing with hybrids and MPI.

Reply:
The error is buried somewhere in the `denseSubtractDensityOfAtomsCmplxNonperiodicBlacs()` routine. One of the reasons seems to be the `scale` factor, which should not be `0.5` in the spin-polarized case, as the time-propagation module seems to follow different conventions than the rest of the code base. But there is more to it, I'm still debugging...

Reply:
See https://github.com/mberdakin/dftbplus/pull/1.

Reply:
I don't remember if in the rest of the code it was like this, but in the real-time module the spin index of the density matrix means up-down, instead of q-m. That is why when we enter the Hamiltonian routine we run ud2qm() for the Mulliken populations.

Reply:
> See [mberdakin#1](https://github.com/mberdakin/dftbplus/pull/1).

I hadn't seen you had the solution, thanks!